File: Backend/ms-pdf/src/main/java/org/example/mspdf/entity/CamposExtraidos.java
package org.example.mspdf.entity;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "campos_extraidos")

public class CamposExtraidos {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nombre;
    @Lob
    @Basic(fetch = FetchType.LAZY) // recomendable para no cargar siempre el BLOB
    private byte[] contenido;
    private String dni;
    private String codigo;
    private String asunto;
    private String identificador;

    @OneToOne
    @JoinColumn(name = "documento_id")
    private DocumentoPDF documentoPDF;

    private String nombreDocumento;
}

File: matdash-free-angular-v3/main/src/app/models/campos.extraidos.ts
import {DocumentoPDF} from "./documentopdf";

export interface CamposExtraidos {
  id?: number;
  nombre?: string;
  dni?: string;
  codigo?: string;
  asunto?: string;
  identificador?: string;
  documentoPDF?: DocumentoPDF;
  nombreDocumento?: string;
}
